src/System.Runtime/src/System/HashCode.cs

```diff
@@ -0,0 +1,406 @@
+                throw new ArgumentNullException(nameof(comparer));
+            Add(comparer.GetHashCode(value));
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private void Add(int value)
+        {
+            unchecked
+            {
+                var val = (uint)value;
+                var position = _length & 0x3;
+
+                switch (position)
+                {
+                    case 0:
+                        _queue1 = val;
+                        break;
+                    case 1:
+                        _queue2 = val;
+                        break;
+                    case 2:
+                        _queue3 = val;
+                        break;
+                    default:
+
+                        var sentinel =
+                            _v1 | _v2 | _v3 | _v4 | // any of accumulators set?
+                            (_length ^ position); // length greater than 3?
+
+                        if (sentinel == 0)
+                            Initialize(out _v1, out _v2, out _v3, out _v4);
+
+                        _v1 = Round(_v1, _queue1);
+                        _v2 = Round(_v2, _queue2);
+                        _v3 = Round(_v3, _queue3);
+                        _v4 = Round(_v4, val);
+
+                        break;
+                }
+
+                _length++;
+            }
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public int ToHashCode()
+        {
+            unchecked
+            {
+                var position = _length & 0x3;
+
+                // If the length is less than 3, _v1 -> _v4 don't contain
+                // anything yet. xxHash32 treats this differently.
+
+                var hash = (_length ^ position) == 0
+                    ? MixEmptyState()
+                    : MixState(_v1, _v2, _v3, _v4);
+
+                hash += _length * 4;
+
+                // Mix what remains in the queue
+
+                switch (position)
+                {
+                    case 1:
+                        hash = QueueRound(hash, _queue1);
+                        break;
+
+                    case 2:
+                        hash = QueueRound(hash, _queue1);
+                        hash = QueueRound(hash, _queue2);
+                        break;
+
+                    case 3:
+                        hash = QueueRound(hash, _queue1);
+                        hash = QueueRound(hash, _queue2);
+                        hash = QueueRound(hash, _queue3);
+                        break;
+                }
+
+                // Final mix
+
+                hash = MixFinal(hash);
+
+                return (int)hash;
+            }
+        }
+
+#pragma warning disable 0809
+
+        [Obsolete("Use ToHashCode to retrieve the computed hash code.", error: true)]
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public override int GetHashCode() => throw new NotImplementedException();
```

Comment:
@terrajobst , do we have a standard exception for this? Perhaps NotSupportedException? We typically wouldn't use NotImplementedException for an API we don't want people to use.